java-logging-spring/src/test/java/uk/gov/hmcts/reform/logging/spring/RequestLoggingComponentTest.java

```diff
@@ -50,24 +52,41 @@ public void listenForEvents() throws JoranException, IOException {
     @Test
     public void requestProcessedMessageShouldBeLoggedForPublicResource() throws Exception {
         ResponseEntity<String> response = restTemplate.getForEntity("/public", String.class);
+
         assertThat(response.getBody()).isEqualTo("OK");
-        assertThat(loggedEvents()).extracting("message").containsOnlyOnce("Request processed");
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        Thread.sleep(10);// how to fix this? test fails to see the log message otherwise
+
+        assertThat(loggedEvents().stream().filter(
+            event -> event.getFormattedMessage().startsWith("Request GET /public processed")
+        ).collect(Collectors.toList())).size().isEqualTo(1);
     }
 
     @Test
     public void requestProcessedMessageShouldBeLoggedForProtectedResource() throws Exception {
         // making sure our Filters are placed outside SecurityFilterChain
         // and get executed no matter request is allowed or not
         ResponseEntity<String> response = restTemplate.getForEntity("/protected", String.class);
+
         assertThat(response.getBody()).isNotEqualTo("OK");
-        assertThat(loggedEvents()).extracting("message").containsOnlyOnce("Request processed");
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
+
+        assertThat(loggedEvents().stream().filter(
+            event -> event.getFormattedMessage().startsWith("Request GET /protected processed")
+        ).collect(Collectors.toList())).size().isEqualTo(1);
```

Comment:
Why changing styles from assertj to hamcrest? This looks very verbose in comparison to the replaced approach.

Reply:
could not regex match against the list of strings. the test case is now replaced with the previous suggestion where i await of logged event to happen